packages/node/src/integrations/local-variables/worker.ts

```diff
@@ -123,7 +115,20 @@ async function handlePaused(
     }
   }
 
-  parentPort?.postMessage({ exceptionHash, frames });
+  // We write the local variables to a property on the error object. These can be read by the integration as the error
+  // event pass through the SDK event pipeline
+  await session.post('Runtime.callFunctionOn', {
+    functionDeclaration: `function() { this.${LOCAL_VARIABLES_KEY} = ${JSON.stringify(frames)}; }`,
+    silent: true,
+    objectId: errorObjectId,
+  });
```

Comment:
I have one question - now that we are doing more inspector work _while the main application is paused_, do we have an idea of how much time awaiting on this `Runtime.callFunctionOn` takes? 

I'm concerned that we might block the main thread (and by extension, the whole event loop) for long enough to be noticeable. I don't know how inspector performance scales with application size either, so it's possible it'd be more impactful on large codebases?

At minimum, is there an approach you recommend that would let us track how long we spend 'paused' before we resume execution? In my normal apps I'd maintain a prometheus metric here or something.
And further, if there's an approach that might let us do the session.post asynchronously and then have the sentry code wait for the data? Increases complexity but might fix this problem. Not worth doing before we know the performance impact of awaiting here, unless there's a good approach you can think of.

Reply:
The memory test code catches an error and captures it's local variables once every millisecond. `console.time` suggests that the debugger is paused for 0.3-0.4ms on my machine to do this.

Whether this has an impact on your app will likely depend on a lot of variables. We have rate limiting for caught exceptions and `maxExceptionsPerSecond` defaults to 50. Rate limiting initially disables capture for 5 seconds and every subsequent rate limit trigger doubles this wait time. We have no rate limiting for uncaught exceptions.

Reply:
Gotcha. If we're pausing on the order of a low single digit milliseconds I'm sure for 99% of use cases this is fine. at 50 exceptions per second and 1ms per exception that blocks the event loop for only 50ms.
At 10ms, that's 500ms blocked.
At 20ms that's an entire second blocked. Which would even prevent us from hitting the 50 exceptions per second 😅 

So yeah as long as we keep execution time low then it's more than fine.
Is it maybe worth adding a WARN log line if any single error execution takes longer than eg. 5ms? That way if people are debugging their slow applications they would be able to see it's the fault of the debugger? 

In general, I just want to find a way to avoid the case where turning this feature on introduces a huge performance problem in some edge case and **application develops have no idea why**. That can lead to a lot of frustration when they eventually realize it's from an observability SDK. If we can expose runtime information _somehow_ that solves my worry.

Thoughts?

Reply:
Yep, it might be a good idea to change our rate-limiting to be based around total blocked time rather than number of paused events

Reply:
I think the happy middleground is some kind of dynamic backpressure detector, so we can adjust rate-limiting according to the volume of exceptions accordingly.

In our server-side SDKs (not node yet though), we do basic heuristic analysis to downsample outgoing events based on relative volume. Spec here: https://develop.sentry.dev/sdk/telemetry/traces/backpressure/. I assume we can take the larger general concept and apply it here.

Let's give this a try afterwards, we can scope it out further.